vi/news/_posts/2016-12-25-ruby-2-4-0-released.md

```diff
@@ -0,0 +1,127 @@
+
+Ruby 2.4.0 là phiên bản stable đầu tiên của Ruby 2.4.0 gồm nhiều tín năng mới như:
+
+## [Giới thiệu cải tiến hash table (bởi Vladimir Makarov)](https://bugs.ruby-lang.org/issues/12142)
+
+Cải thiện cấu trúc bên trong của hash table (st_table)
+Improve the internal structure of hash table (st_table) bằng việc giới thiệu
+open addressing và một mảng bao hàm có thứ tự.
+Cải thiện này đã được rất nhiều người chấp nhận, đặc biệt là Yura Sokolov.
+
+## Binding#irb: Bắt đầu một REPL session tương tự như `binding.pry`
+
+Thông thường bạn có thể dùng `p` để xem giá trị của các biến trong quá trình debug.
+Với [pry](https://github.com/pry/pry) bạn có thể dùng `binding.pry` để chạy REPL và
+debug ở bất kỳ dòng code nào.
+[r56624](https://github.com/ruby/ruby/commit/493e48897421d176a8faf0f0820323d79ecdf94a) giới thiệu `binding.irb` hoạt động tương tự IRB.
+
+## [Hợp nhất Fixnum và Bignum vào Integer](https://bugs.ruby-lang.org/issues/12005)
+
+Mặc dù chuẩn [ISO/IEC 30170:2012](http://www.iso.org/iso/iso_catalogue/catalogue_tc/catalogue_detail.htm?csnumber=59579)
+không mô tả chi tiết nội dung của lớp Integer,
+Ruby hiện đang có hai lớp Integer: Fixnum và Bignum.
+Ruby 2.4 gộp hai lớp này thành một lớp là Integer.
+Tất cả các extensions của C mà có liên quan đến lớp Fixnum hoặc Bignum
+thì cần được thay đổi.
+
+Xem thêm [the ticket](https://bugs.ruby-lang.org/issues/12005) và [akr's slides](http://www.a-k-r.org/pub/2016-09-08-rubykaigi-unified-integer.pdf).
+
+## [Class String hỗ trợ Unicode trong xử lí chữ hoa, chữ thường](https://bugs.ruby-lang.org/issues/10085)
+
+Hàm `String/Symbol#upcase/downcase/swapcase/capitalize(!)` bây giờ đã có thể xử lý hoa-thường cho cả Unicode (thay vì chỉ cho ASCII).
+
+## Các cải thiện về hiệu năng
+
+Ruby 2.4 cũng bao gồm các cải thiện về hiệu năng như sau:
+
+### [Array#max, Array#min](https://bugs.ruby-lang.org/issues/12172)
+
+`[x, y].max` và `[x, y].min` đã được cải tiến để không tạo ra một mảng tạm trong một số trường hợp.
+
+### [Regexp#match?](https://bugs.ruby-lang.org/issues/8110)
+
+Hàm `Regexp#match?` đã được thêm vào với chức năng thực hiện việc kiểm tra một
+biểu thức chính quy mà không tạo ra một đối tượng cho các tham chiếu ngược lai
+và thay đổi `$~` nhằm giảm việc khởi tạo các đối tượng.
+
+### Những sự cải thiện về hiệu năng khác
+
+* [Tăng tốc độ truy cập biến instance](https://bugs.ruby-lang.org/issues/12274)
+
+## Debugging
+
+### [Thread#report_on_exception and Thread.report_on_exception](https://bugs.ruby-lang.org/issues/6647)
+
+Ruby sẽ bỏ qua các exception trong các thread nếu không có một thread khác tham
+gia vào tiến trình của thread có exception. Tuy nhiên với việc set `report_on_exception = true`,
+bạn có thể nhận biết được việc thread bị kết thúc bởi một exception chưa được xử lý.
+
+Gửi góp ý cho chúng tôi về giá trị mặc định của `report_on_exception`
+và report-on-GC (thống kê ở Garbage Collection), đưa ra một báo cáo khi một
```

Comment:
This might cause confusion, maybe we could translate it to:

Gửi góp ý cho chúng tôi về giá trị mặc định của `report_on_exception` và report-on-GC (tính năng sẽ đưa ra báo cáo khi một thread được dọn dẹp mà không tham gia).